packages/core/test/features/css-class.spec.ts

```diff
@@ -1,483 +1,718 @@
+        it(`should report unscoped class`, () => {
+            /*
+            ToDo: consider to accept as scoped when local symbol exists
+            anywhere in the selector: ".importedPart .local div"
+            */
+            const { sheets } = testStylableCore({
+                '/classes.st.css': `
+                    .importedPart {}
+                `,
+                '/entry.st.css': `
+                    @st-import [importedPart] from "./classes.st.css";
 
-                testInlineExpects(result);
+                    /* @analyze-warn word(importedPart) ${CSSClass.diagnostics.UNSCOPED_CLASS(
+                        `importedPart`
+                    )} */
+                    .importedPart {}
+
+                    /* NO ERROR - locally scoped */
+                    .local .importedPart {}
+                    .local.importedPart {}
+                    .importedPart.local {}
+                `,
             });
+
+            const { meta } = sheets[`/entry.st.css`];
+
+            expect(meta.diagnostics.reports.length, `only unscoped diagnostic`).to.equal(1);
         });
-    });
-    describe(`diagnostics`, () => {
-        it(`should error on unsupported functional class`, () => {
-            expectAnalyzeDiagnostics(
-                `|$.abc()$| {}`,
-                [
-                    {
-                        file: `/entry.st.css`,
-                        message: CSSClass.diagnostics.INVALID_FUNCTIONAL_SELECTOR(`.abc`, `class`),
-                        severity: `error`,
-                    },
-                ],
-                { partial: true }
-            );
+        it(`should override with imported -st-global`, () => {
+            const { sheets } = testStylableCore({
+                '/comp.st.css': `
+                    .root {
+                        -st-global: .r;
+                    }
+                    .part {
+                        -st-global: .p;
+                    }
+                `,
+                '/entry.st.css': `
+                    @st-import Comp, [root as iRoot, part as iPart] from './comp.st.css';
+
+                    /* @rule .r */
+                    Comp {}
+                    
+                    /* @rule .r */
+                    .iRoot {}
+
+                    /* @rule .p */
+                    .iPart {}
+                `,
+            });
+
+            const { meta, exports } = sheets['/entry.st.css'];
+
+            // symbols
+            expect(CSSClass.get(meta, `iRoot`), `iRoot symbol`).to.contain({
+                name: 'iRoot',
+            });
+            expect(CSSClass.get(meta, `iPart`), `iPart symbol`).to.contain({
+                name: 'iPart',
+            });
+
+            // meta.globals
+            expect(meta.globals).to.eql({
+                r: true,
+                p: true,
+            });
+
+            // JS exports
+            expect(exports.classes.iRoot, `root alias JS export`).to.eql(undefined);
```

Comment:
chack d.ts for alias root

Reply:
hasOwnProperty?

Reply:
`d.ts` usees exports to generate the type, so at the before this change the `d.ts` created an key for the aliased root, although the value is an empty string in reality, so this change will fix this issue as well - imported aliased root will not show as key under exported JS classes or in the `d.ts`